cloud/aws/services/ec2/vpc.go

```diff
@@ -17,41 +17,85 @@ import (
 	"errors"
 	"fmt"
 
+	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/service/ec2"
+	"sigs.k8s.io/cluster-api-provider-aws/cloud/aws/providerconfig/v1alpha1"
 )
 
 type vpcs interface {
-	DescribeVpcs(*ec2.DescribeVpcsInput) (*ec2.DescribeVpcsOutput, error)
+	DescribeVpcs(input *ec2.DescribeVpcsInput) (*ec2.DescribeVpcsOutput, error)
+	CreateVpc(input *ec2.CreateVpcInput) (*ec2.CreateVpcOutput, error)
+	DeleteVpc(input *ec2.DeleteVpcInput) (*ec2.DeleteVpcOutput, error)
 }
 
 type VPC struct {
-	ID string
+	ID        string
+	CidrBlock string // TODO(vincepri): use *net.IPNet?
 }
 
-func (s *Service) ReconcileVPC(id string) (*VPC, error) {
+func (s *Service) ReconcileVPC(v *v1alpha1.VPC) (*VPC, error) {
 	// Does it exist and look in good working order? ok exit no error
-	vpc, err := s.lookupVPCByID(id)
+	vpc, err := s.lookupVPCByID(v.ID)
 	if err != nil {
+		if err == ErrVPCNotFound {
+			return s.createVPC(v)
+		}
+
 		return nil, err
 	}
-	// If it doesn't exist, create it
+
+	if v.CidrBlock != vpc.CidrBlock {
+		// TODO(vincepri): delete and recreate?
```

Comment:
hmm, deleting and recreating the VPC doesn't sound like an ideal thing...maybe we hard error if someone tries to update a field that is not updatable? TODO is ok for now, but let's think about it/track it

Reply:
I think we should probably have a flag to allow the VPC to be re-created. To delete a VPC each single item inside of it needs to be cleaned up first, for example instances, gateways, etc. 

Another thing to consider is that VPCs can now have additional cidr blocks, which are added through additional calls.